supernode/services/artworkregister/task.go

```diff
@@ -179,20 +182,57 @@ func (task *Task) ProbeImage(_ context.Context, file *artwork.File) ([]byte, err
 		}
 
 		// NOTE: for testing Kademlia and should be removed before releasing.
-		data, err := file.Bytes()
+		// data, err := file.Bytes()
+		// if err != nil {
+		// 	return err
+		// }
+
+		// id, err := task.p2pClient.Store(ctx, data)
+		// if err != nil {
+		// 	return err
+		// }
+		// log.WithContext(ctx).WithField("id", id).Debugf("Image stored into Kademlia")
+		return nil
+	})
+
+	return fingerprintData, nil
+}
+
+// UploadImageWithThumbnail uploads the image that contained image with pqsignature
+// generate the image thumbnail from the coordinate provided for user and return
+// the hash for the genreated thumbnail
+func (task *Task) UploadImageWithThumbnail(_ context.Context, file *artwork.File, thumbnail artwork.ImageThumbnail) ([]byte, error) {
+	if err := task.RequiredStatus(StatusImageProbed); err != nil {
+		return nil, err
```

Comment:
please wrap with detail info. Like  errors.Errorf("required state is not statisified %w", err)